### PR TITLE
(APS-602) Return task outcome and recorded at date

### PR DESCRIPTION
**Pull request by pezholio**
Now we’re returning completed tasks, we need to record what the outcome was and when that outcome was recorded.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/PlacementApplicationEntity.kt [hunk 1]
```diff
@@ -153,6 +155,14 @@ enum class PlacementApplicationDecision {
   REJECTED,
   WITHDRAW,
   WITHDRAWN_BY_PP,
+  ;
+
+  fun convertToApi() = when (this) {
```
Comment (davidatkinsuk) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/PlacementApplicationEntity.kt:160]: nice

**Review by davidatkinsuk — APPROVED** [review]
LGTM 👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/PlacementApplicationEntity.kt (hunk 1 above)
Comment (pezholio) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/PlacementApplicationEntity.kt:160]: I think we could do this with a lot of our enum classes

Comment (davidatkinsuk) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/PlacementApplicationEntity.kt:160]: I think we could do something like this and then add a function to do the inverse mapping if required (possibly as an extension function with kotlin enums can implement interfaces. Java is inflexible around enums

```
enum class PlacementApplicationDecision(val ApiPlacementApplicationDecision apiDecision) {
  ACCEPTED(accepted),
  REJECTED(rejected),
  WITHDRAW(withdraw),
  WITHDRAWN_BY_PP(withdrawnByPp),
}
```

**Pull request merged**